subt_ign/src/path_tracer.cc

```diff
@@ -200,13 +200,25 @@ void Processor::ArtifactCb(const ignition::msgs::Pose_V &_msg)
 //////////////////////////////////////////////////
 void Processor::DisplayPoses()
 {
-  for (auto &stepData : this->logData)
+  for (std::map<int, std::vector<std::unique_ptr<Data>>>::iterator iter =
+       this->logData.begin(); iter != this->logData.end(); ++iter)
   {
-    for (std::unique_ptr<Data> &data : stepData.second)
+    printf("\r %ds/%ds (%06.2f%%)", iter->first, this->logData.rbegin()->first,
+        static_cast<double>(iter->first) / this->logData.rbegin()->first * 100);
+    fflush(stdout);
+
+    for (std::unique_ptr<Data> &data : iter->second)
     {
       data->Render(this);
     }
-    std::this_thread::sleep_for(std::chrono::milliseconds(this->stepSleepMs));
+
+    // Get the next time stamp, and sleep the correct amount of time.
+    auto next = std::next(iter, 1);
+    if (next != this->logData.end())
+    {
+      int sleepTime = ((next->first - iter->first) / this->rtf)*1000;
```

Comment:
if we want to be accurate, we can set minus `sleepTime` by the amount of time it's already taken in this iteration. Not really necessary if the `for` loop with `Render` operation is relatively fast.

Reply:
Good idea. I added it in here: b4adf4529d18f53a3793cc73984bc16dc22540a6.